backend/layers/business/business.py

```diff
@@ -822,13 +824,14 @@ def _delete_from_bucket(self, bucket: str, keys: List[str] = None, prefix: str =
             if keys:
                 self.s3_provider.delete_files(bucket, keys)
             if prefix:
-                self.s3_provider.delete_recursive(bucket, prefix)
+                object_keys = list(self.s3_provider.list_directory(bucket, prefix))
```

Comment:
nit: I would suggest pushing these to functions into another function in s3_provider. Maybe call it `delete_prefixed_files(bucket, prefix)`. It seems like an operation that could be useful to other areas of the code.

Reply:
incorporated 👍